### PR TITLE
Eugene_functions.js done

### PR DESCRIPTION
**Pull request by yevhensydorov**

### CODE REVIEW

File: src/functions.js [hunk 1]
```diff
@@ -3,31 +3,42 @@ function removeMiddle( words ){
...
 
 function mapper( myArray ){
   // myArray is an array of numbers
   // return a new array which has all items in myArray incremented by one
   // myArray should remain unchanged
   // hint: map
+  var incremented = myArray.map(function(a){
+    return a += 1;
+  });
+  return incremented;
```
Comment (alismith-dev) [inline src/functions.js]: Is there a need for the `incremented` variable?

File: src/functions.js [hunk 2]
```diff
@@ -3,31 +3,42 @@ function removeMiddle( words ){
...
 }
 
 function wordLengths( words ){
   // words is an array of strings
   // return a new array that contains the number of letters in each word
   // hint: strings have a 'length' property
+  var num = [];
+  words.forEach(function(arr){
```
Comment (alismith-dev) [inline src/functions.js]: Is there a way you could make this more concise? Note that you're taking an existing array and using it to create a new array. This pattern is so common that it's been added as a native array function.

File: src/functions.js [hunk 3]
```diff
@@ -109,13 +148,29 @@ function sales( cars ){
...
 // Harder challenges
 function secondLargest( numbers ){
   // numbers is an array of numbers
   // return the index of the second 
   // largest number in the array
+  var max = numbers.reduce(function(a, b) {return Math.max(a, b)});
+  numbers.splice(numbers.indexOf(max), 1);
+  var secondMax = numbers.reduce(function(a, b) {return Math.max(a, b)});
+
+  for(var i = 0; i < numbers.length; i++){
```
Comment (alismith-dev) [inline src/functions.js]: This is a slightly inefficient way of acheiving this. `return numbers.indexOf(secondMax)` I think would work too

**Review by alismith-dev — COMMENTED** [review]
Nice work 😄 Left a couple of comments about some improvements you could make

File: src/functions.js (hunk 2 above)
Comment (yevhensydorov) [reply src/functions.js]: yes, done with map method!

File: src/functions.js (hunk 1 above)
Comment (yevhensydorov) [reply src/functions.js]: made this a bit shortly

File: src/functions.js (hunk 3 above)
Comment (yevhensydorov) [reply src/functions.js]: Thanks Ali!

**Pull request closed without merging**